File: server/php/src/Infrastructure/Mappers/PaymentDetails/DiscountMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\Discount as DiscountDto;
use OnlinePayments\Sdk\Domain\Discount as DiscountSdk;

final class DiscountMapper
{
    public static function mapFromSdkResponse(?DiscountSdk $response): ?DiscountDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new DiscountDto();
        $dto->amount = $response?->amount ?? null;

        return $dto;
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/DiscountMapper.cs
using DiscountSdk = OnlinePayments.Sdk.Domain.Discount;
using DiscountDto = Business.Domain.Payments.PaymentDetails.Discount;

namespace Infrastructure.Mappers.PaymentDetails;

public static class DiscountMapper
{
    public static DiscountDto Map(DiscountSdk? response)
    {
        return new()
        {
            Amount = response?.Amount
        };
    }
}